python/HeaderService/hslib_salobj.py

```diff
@@ -0,0 +1,500 @@
+        emulate it for the EFD
+        """
+        # Get the md5 for the header file
+        md5value = hutils.md5Checksum(self.filename_HDR)
+        bytesize = os.path.getsize(self.filename_HDR)
+        LOGGER.info("Got MD5SUM: {}".format(md5value))
+        # Now we publish filename and MD5
+        # Build the kwargs
+        kw = {'byteSize': bytesize,
+              'checkSum': md5value,
+              'generator': self.hs_name,
+              'mimeType': self.write_mode.upper(),
+              'url': self.url_format.format(ip_address=self.ip_address,
+                                            port_number=self.port_number,
+                                            filename_HDR=os.path.basename(self.filename_HDR)),
+              'id': self.imageName,
+              'version': 1,
+              'priority': 1,
+              }
+
+        self.hs_controller.evt_largeFileObjectAvailable.set_put(**kw)
+        LOGGER.info(f"Sent {self.hs_name} largeFileObjectAvailable: {kw}")
+        if self.send_efd_message:
+            self.efd_controller.evt_largeFileObjectAvailable.set_put(**kw)
+            LOGGER.info(f"Sent EFD largeFileObjectAvailable: {kw}")
+
+    def write(self):
+        """ Function to call to write the header"""
+        self.HDR.write_header(self.filename_HDR)
+        LOGGER.info("Wrote header to: {}".format(self.filename_HDR))
+
+    def clean(self):
+        """ Clean up metadata and myData dicts"""
+        self.myData = {}
+        self.metadata = {}
+
+    def collect(self, keys):
+        """ Collect meta-data from the telemetry-connected channels
+        and store it in the 'metadata' dictionary"""
+
+        # Make sure the myData payload is reset before collection
+        # to clean up data from previous events/telemetry
+        self.myData = {}
+        for k in keys:
+            name = get_channel_name(self.telemetry[k])
+            param = self.telemetry[k]['value']
+            # Access data payload only once
+            if name not in self.myData:
+                self.myData[name] = self.Remote_get[name]()
+                # Only update metadata if self.myData is defined (not None)
+            if self.myData[name] is None:
+                LOGGER.warning(f"Cannot get keyword: {k} from topic: {name}")
+            else:
+                self.metadata[k] = getattr(self.myData[name], param)
+                LOGGER.debug(f"Extacted {k}={self.metadata[k]} from topic: {name}")
+
+    def collect_from_HeaderService(self):
+
+        """
+        Collect and update custom meta-data generated or transformed by
+        the HeaderService
+        """
+
+        # Reformat and calculate dates based on different timeStamps
+        self.DATE = hscalc.get_date_utc(self.metadata['DATE'])
+        self.DATE_OBS = hscalc.get_date_utc(self.metadata['DATE-OBS'])
+        self.DATE_BEG = hscalc.get_date_utc(self.metadata['DATE-BEG'])
+        self.DATE_END = hscalc.get_date_utc(self.metadata['DATE-END'])
```

Comment:
I am pretty sure LSST chose to use TAI for all timestamps. In which case these should be TAI, not UTC and you should set FITS header `TIMESYS=TAI`.

Reply:
As suggested by Tim J, we are using UTC times for the FITS headers at the moment.

Reply:
LSE-400 says we should use TIMESYS=TAI so we should be switching over.

Reply:
Thanks for the clarification, will update accordingly.